cairo-contracts/packages/apps/src/transfer/components/transfer.cairo

```diff
@@ -283,19 +283,30 @@ pub mod TokenTransferComponent {
 
     #[embeddable_as(CreateIbcToken)]
     impl CreateTokenImpl<
-        TContractState, +HasComponent<TContractState>, +Drop<TContractState>,
+        TContractState,
+        +HasComponent<TContractState>,
+        +Drop<TContractState>,
+        impl Ownable: OwnableComponent::HasComponent<TContractState>,
     > of ICreateIbcToken<ComponentState<TContractState>> {
         fn create_ibc_token(
-            ref self: ComponentState<TContractState>,
-            chan_id_on_b: ChannelId,
-            base_denom: ByteArray,
+            ref self: ComponentState<TContractState>, denom: PrefixedDenom,
         ) -> ContractAddress {
-            let token = self.create_token(base_denom.clone(), 0);
-            let trace_prefix = TracePrefixTrait::new(TRANSFER_PORT_ID(), chan_id_on_b);
-            let prefixed_denom = PrefixedDenom {
-                trace_path: array![trace_prefix], base: Denom::Hosted(base_denom),
-            };
-            self.record_ibc_token(prefixed_denom, token.address);
+            let channel: ChannelContract = self.owner().into();
+
+            let maybe_prefix = denom.last_prefix();
+            assert(maybe_prefix.is_some(), TransferErrors::MISSING_TRACE_PREFIX);
+            let prefix = maybe_prefix.unwrap();
+
+            let chan_end_on_b = channel
+                .channel_end(prefix.port_id.clone(), prefix.channel_id.clone());
```

Comment:
does it make sense to check if `prefix.port_id` is `"transfer"` ?

Reply:
It's actually checked implicitly. But not bad, for the sake of better error reporting to have this aux check too:
[d6203ed](https://github.com/informalsystems/ibc-starknet/pull/326/commits/d6203edef51dadc518d2fea3a253fe9d99ebce51)

Reply:
Interesting! can you show where it is checked implicitly?

Reply:
The channel end won't exist if anything else than `transfer` passed.